File: src/bindings/ruby/test/TestSpecies.rb
#
# @file    TestSpecies.rb
# @brief   Species unit tests
# @author  Akiya Jouraku (Ruby conversion)
# @author  Ben Bornstein 
#
# $Id$
# $Source$
#
# This test file was converted from src/sbml/test/TestSpecies.c
# wiht the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2008 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
#
require 'test/unit'
require 'libSBML'

class TestSpecies < Test::Unit::TestCase

  def setup
    @@s = LibSBML::Species.new
    if (@@s == nil)
    end
  end

  def test_Species_create
    assert( @@s.getTypeCode == LibSBML::SBML_SPECIES )
    assert( @@s.getMetaId == "" )
    assert( @@s.getNotes == nil )
    assert( @@s.getAnnotation == nil )
    assert( @@s.getId == "" )
    assert( @@s.getName == "" )
    assert( @@s.getCompartment == "" )
    assert( @@s.getInitialAmount == 0.0 )
    assert( @@s.getInitialConcentration == 0.0 )
    assert( @@s.getSubstanceUnits == "" )
    assert( @@s.getSpatialSizeUnits == "" )
    assert( @@s.getHasOnlySubstanceUnits == false )
    assert( @@s.getBoundaryCondition == false )
    assert( @@s.getCharge == 0 )
    assert( @@s.getConstant == false )
    assert_equal false, @@s.isSetId
    assert_equal false, @@s.isSetName
    assert_equal false, @@s.isSetCompartment
    assert_equal false, @@s.isSetInitialAmount
    assert_equal false, @@s.isSetInitialConcentration
    assert_equal false, @@s.isSetSubstanceUnits
    assert_equal false, @@s.isSetSpatialSizeUnits
    assert_equal false, @@s.isSetUnits
    assert_equal false, @@s.isSetCharge
  end

  def test_Species_createWith
    s = LibSBML::Species.new("Ca", "Calcium")
    assert( s.getTypeCode == LibSBML::SBML_SPECIES )
    assert( s.getMetaId == "" )
    assert( s.getNotes == nil )
    assert( s.getAnnotation == nil )
    assert ((  "Calcium"   == s.getName ))
    assert( s.getSpatialSizeUnits == "" )
    assert( s.getHasOnlySubstanceUnits == false )
    assert( s.getConstant == false )
    assert ((  "Ca"   == s.getId ))
    assert_equal true, s.isSetId
    assert_equal true, s.isSetName
    assert_equal false, s.isSetCompartment
    assert_equal false, s.isSetSubstanceUnits
    assert_equal false, s.isSetSpatialSizeUnits
    assert_equal false, s.isSetUnits
    assert_equal false, s.isSetInitialAmount
    assert_equal false, s.isSetInitialConcentration
    assert_equal false, s.isSetCharge
  end

  def test_Species_free_NULL
  end

  def test_Species_setCompartment
    compartment = "cell"
    @@s.setCompartment(compartment)
    assert (( compartment == @@s.getCompartment ))
    assert_equal true, @@s.isSetCompartment
    if (@@s.getCompartment == compartment)
    end
    @@s.setCompartment(@@s.getCompartment)
    assert (( compartment == @@s.getCompartment ))
    @@s.setCompartment("")
    assert_equal false, @@s.isSetCompartment
    if (@@s.getCompartment != nil)
    end
  end

  def test_Species_setId
    id = "Glucose"
    @@s.setId(id)
    assert (( id == @@s.getId ))
    assert_equal true, @@s.isSetId
    if (@@s.getId == id)
    end
    @@s.setId(@@s.getId)
    assert (( id == @@s.getId ))
    @@s.setId("")
    assert_equal false, @@s.isSetId
    if (@@s.getId != nil)
    end
  end

  def test_Species_setInitialAmount
    assert_equal false, @@s.isSetInitialAmount
    assert_equal false, @@s.isSetInitialConcentration
    @@s.setInitialAmount(1.2)
    assert_equal true, @@s.isSetInitialAmount
    assert_equal false, @@s.isSetInitialConcentration
    assert( @@s.getInitialAmount == 1.2 )
  end

  def test_Species_setInitialConcentration
    assert_equal false, @@s.isSetInitialAmount
    assert_equal false, @@s.isSetInitialConcentration
    @@s.setInitialConcentration(3.4)
    assert_equal false, @@s.isSetInitialAmount
    assert_equal true, @@s.isSetInitialConcentration
    assert( @@s.getInitialConcentration == 3.4 )
  end

  def test_Species_setName
    name = "So Sweet"
    @@s.setName(name)
    assert (( name == @@s.getName ))
    assert_equal true, @@s.isSetName
    if (@@s.getName == name)
    end
    @@s.setName(@@s.getName)
    assert (( name == @@s.getName ))
    @@s.setName("")
    assert_equal false, @@s.isSetName
    if (@@s.getName != nil)
    end
  end

  def test_Species_setSpatialSizeUnits
    units = "volume"
    @@s.setSpatialSizeUnits(units)
    assert (( units == @@s.getSpatialSizeUnits ))
    assert_equal true, @@s.isSetSpatialSizeUnits
    if (@@s.getSpatialSizeUnits == units)
    end
    @@s.setSpatialSizeUnits(@@s.getSpatialSizeUnits)
    assert (( units == @@s.getSpatialSizeUnits ))
    @@s.setSpatialSizeUnits("")
    assert_equal false, @@s.isSetSpatialSizeUnits
    if (@@s.getSpatialSizeUnits != nil)
    end
  end

  def test_Species_setSubstanceUnits
    units = "item"
    @@s.setSubstanceUnits(units)
    assert (( units == @@s.getSubstanceUnits ))
    assert_equal true, @@s.isSetSubstanceUnits
    if (@@s.getSubstanceUnits == units)
    end
    @@s.setSubstanceUnits(@@s.getSubstanceUnits)
    assert (( units == @@s.getSubstanceUnits ))
    @@s.setSubstanceUnits("")
    assert_equal false, @@s.isSetSubstanceUnits
    if (@@s.getSubstanceUnits != nil)
    end
  end

  def test_Species_setUnits
    units = "mole"
    @@s.setUnits(units)
    assert (( units == @@s.getUnits ))
    assert_equal true, @@s.isSetUnits
    if (@@s.getSubstanceUnits == units)
    end
    @@s.setUnits(@@s.getSubstanceUnits)
    assert (( units == @@s.getUnits ))
    @@s.setUnits("")
    assert_equal false, @@s.isSetUnits
    if (@@s.getSubstanceUnits != nil)
    end
  end

end

File: src/bindings/python/test/TestSpecies.py
#
# @file    TestSpecies.py
# @brief   Species unit tests
# @author  Akiya Jouraku (Python conversion)
# @author  Ben Bornstein 
#
# $Id$
# $Source$
#
# This test file was converted from src/sbml/test/TestSpecies.c
# with the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2008 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
#
import sys
import unittest
import libsbml

class TestSpecies(unittest.TestCase):

  S = None

  def setUp(self):
    self.S = libsbml.Species()
    if (self.S == None):
      pass    
    pass  

  def tearDown(self):
    self.S = None
    pass  

  def test_Species_create(self):
    self.assert_( self.S.getTypeCode() == libsbml.SBML_SPECIES )
    self.assert_( self.S.getMetaId() == "" )
    self.assert_( self.S.getNotes() == None )
    self.assert_( self.S.getAnnotation() == None )
    self.assert_( self.S.getId() == "" )
    self.assert_( self.S.getName() == "" )
    self.assert_( self.S.getCompartment() == "" )
    self.assert_( self.S.getInitialAmount() == 0.0 )
    self.assert_( self.S.getInitialConcentration() == 0.0 )
    self.assert_( self.S.getSubstanceUnits() == "" )
    self.assert_( self.S.getSpatialSizeUnits() == "" )
    self.assert_( self.S.getHasOnlySubstanceUnits() == False )
    self.assert_( self.S.getBoundaryCondition() == False )
    self.assert_( self.S.getCharge() == 0 )
    self.assert_( self.S.getConstant() == False )
    self.assertEqual( False, self.S.isSetId() )
    self.assertEqual( False, self.S.isSetName() )
    self.assertEqual( False, self.S.isSetCompartment() )
    self.assertEqual( False, self.S.isSetInitialAmount() )
    self.assertEqual( False, self.S.isSetInitialConcentration() )
    self.assertEqual( False, self.S.isSetSubstanceUnits() )
    self.assertEqual( False, self.S.isSetSpatialSizeUnits() )
    self.assertEqual( False, self.S.isSetUnits() )
    self.assertEqual( False, self.S.isSetCharge() )
    pass  

  def test_Species_createWith(self):
    s = libsbml.Species("Ca", "Calcium")
    self.assert_( s.getTypeCode() == libsbml.SBML_SPECIES )
    self.assert_( s.getMetaId() == "" )
    self.assert_( s.getNotes() == None )
    self.assert_( s.getAnnotation() == None )
    self.assert_((  "Calcium"   == s.getName() ))
    self.assert_( s.getSpatialSizeUnits() == "" )
    self.assert_( s.getHasOnlySubstanceUnits() == False )
    self.assert_( s.getConstant() == False )
    self.assert_((  "Ca"   == s.getId() ))
    self.assertEqual( True, s.isSetId() )
    self.assertEqual( True, s.isSetName() )
    self.assertEqual( False, s.isSetCompartment() )
    self.assertEqual( False, s.isSetSubstanceUnits() )
    self.assertEqual( False, s.isSetSpatialSizeUnits() )
    self.assertEqual( False, s.isSetUnits() )
    self.assertEqual( False, s.isSetInitialAmount() )
    self.assertEqual( False, s.isSetInitialConcentration() )
    self.assertEqual( False, s.isSetCharge() )
    s = None
    pass  

  def test_Species_free_NULL(self):
    
    pass  

  def test_Species_setCompartment(self):
    compartment = "cell"
    self.S.setCompartment(compartment)
    self.assert_(( compartment == self.S.getCompartment() ))
    self.assertEqual( True, self.S.isSetCompartment() )
    if (self.S.getCompartment() == compartment):
      pass    
    self.S.setCompartment(self.S.getCompartment())
    self.assert_(( compartment == self.S.getCompartment() ))
    self.S.setCompartment("")
    self.assertEqual( False, self.S.isSetCompartment() )
    if (self.S.getCompartment() != None):
      pass    
    pass  

  def test_Species_setId(self):
    id = "Glucose"
    self.S.setId(id)
    self.assert_(( id == self.S.getId() ))
    self.assertEqual( True, self.S.isSetId() )
    if (self.S.getId() == id):
      pass    
    self.S.setId(self.S.getId())
    self.assert_(( id == self.S.getId() ))
    self.S.setId("")
    self.assertEqual( False, self.S.isSetId() )
    if (self.S.getId() != None):
      pass    
    pass  

  def test_Species_setInitialAmount(self):
    self.assertEqual( False, self.S.isSetInitialAmount() )
    self.assertEqual( False, self.S.isSetInitialConcentration() )
    self.S.setInitialAmount(1.2)
    self.assertEqual( True, self.S.isSetInitialAmount() )
    self.assertEqual( False, self.S.isSetInitialConcentration() )
    self.assert_( self.S.getInitialAmount() == 1.2 )
    pass  

  def test_Species_setInitialConcentration(self):
    self.assertEqual( False, self.S.isSetInitialAmount() )
    self.assertEqual( False, self.S.isSetInitialConcentration() )
    self.S.setInitialConcentration(3.4)
    self.assertEqual( False, self.S.isSetInitialAmount() )
    self.assertEqual( True, self.S.isSetInitialConcentration() )
    self.assert_( self.S.getInitialConcentration() == 3.4 )
    pass  

  def test_Species_setName(self):
    name = "So Sweet"
    self.S.setName(name)
    self.assert_(( name == self.S.getName() ))
    self.assertEqual( True, self.S.isSetName() )
    if (self.S.getName() == name):
      pass    
    self.S.setName(self.S.getName())
    self.assert_(( name == self.S.getName() ))
    self.S.setName("")
    self.assertEqual( False, self.S.isSetName() )
    if (self.S.getName() != None):
      pass    
    pass  

  def test_Species_setSpatialSizeUnits(self):
    units = "volume"
    self.S.setSpatialSizeUnits(units)
    self.assert_(( units == self.S.getSpatialSizeUnits() ))
    self.assertEqual( True, self.S.isSetSpatialSizeUnits() )
    if (self.S.getSpatialSizeUnits() == units):
      pass    
    self.S.setSpatialSizeUnits(self.S.getSpatialSizeUnits())
    self.assert_(( units == self.S.getSpatialSizeUnits() ))
    self.S.setSpatialSizeUnits("")
    self.assertEqual( False, self.S.isSetSpatialSizeUnits() )
    if (self.S.getSpatialSizeUnits() != None):
      pass    
    pass  

  def test_Species_setSubstanceUnits(self):
    units = "item"
    self.S.setSubstanceUnits(units)
    self.assert_(( units == self.S.getSubstanceUnits() ))
    self.assertEqual( True, self.S.isSetSubstanceUnits() )
    if (self.S.getSubstanceUnits() == units):
      pass    
    self.S.setSubstanceUnits(self.S.getSubstanceUnits())
    self.assert_(( units == self.S.getSubstanceUnits() ))
    self.S.setSubstanceUnits("")
    self.assertEqual( False, self.S.isSetSubstanceUnits() )
    if (self.S.getSubstanceUnits() != None):
      pass    
    pass  

  def test_Species_setUnits(self):
    units = "mole"
    self.S.setUnits(units)
    self.assert_(( units == self.S.getUnits() ))
    self.assertEqual( True, self.S.isSetUnits() )
    if (self.S.getSubstanceUnits() == units):
      pass    
    self.S.setUnits(self.S.getSubstanceUnits())
    self.assert_(( units == self.S.getUnits() ))
    self.S.setUnits("")
    self.assertEqual( False, self.S.isSetUnits() )
    if (self.S.getSubstanceUnits() != None):
      pass    
    pass  

def suite():
  suite = unittest.TestSuite()
  suite.addTest(unittest.makeSuite(TestSpecies))

  return suite

if __name__ == "__main__":
  if unittest.TextTestRunner(verbosity=1).run(suite()).wasSuccessful() :
    sys.exit(0)
  else:
    sys.exit(1)
